dockertest/subtestbase.py

```diff
@@ -102,10 +161,84 @@ def cleanup(self):
         Always called, before any exceptions thrown are re-raised.
         """
         self.log_step_msg('cleanup')
+        self.overlook_known_failures()
+
+    def overlook_known_failures(self):
+        """
+        If we've caught a DockerXxxx exception, see if it's a known problem
+        with this combination of subtest & docker NVR. If it is, change the
+        exception status from FAIL to WARN.
+        """
+        e_type, e_value, _ = sys.exc_info()
+
+        # A string comparison lets us catch any DockerXxxxx exception,
+        # e.g. DockerTestFail, DockerOutputError
+        if 'Docker' not in str(e_type):
+            return
+
+        # Children of autotest TestBaseException have an exit_status field;
+        # we only bypass if it's FAIL
+        if not hasattr(e_value, 'exit_status'):
+            return
+        if e_value.exit_status != 'FAIL':
+            return
+
+        # Yep, it's a test failure
+        # FIXME: is there a way to check if this is a new exception
+        #        coming from our module, vs an old one from a previous test?
```

Comment:
To the best of my knowledge, this is incredibly difficult to do because exception-handling can also raise exceptions.  There's also weird things, like if an exception is thrown from an instances ``__del__()`` method, python changes it to a warning.

I'm willing to see how this plays out in reality, what you have here may be good enough.

A more "controlled" way to handle this situation via exceptions could be decorating a few key ``SubBase`` and ``Subtest`` base-methods with ``error.context_aware()`` (see example in ``autotest/client/shared/error.py``).  For example, ``run_once()`` and ``initialize()``.  Then those methods can use ``error.context()`` to apply set some magic string, which will automatically be attached to any raised exceptions.  Finally, this function can then use ``e_value.get_context()`` to validate the magic string to know from whence the exception came 😀

Reply:
The decorator-way (suggestion above) would also address the broken ``isinstance()`` thing.  Just sayin.  Don't run out and do ``@context_aware()`` unless you think it's a good idea.  I'm fine if that's a later addition, or never.

Reply:
Still looking into `@error.context`. Looks really intriguing but I'm not entirely sure how I could take advantage of it. Thanks for the tip.